core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedStageExecution.java

```diff
@@ -559,6 +560,12 @@ public Optional<ExecutionFailureInfo> getFailureCause()
         return stateMachine.getFailureCause();
     }
 
+    @Override
+    public String toString()
+    {
+        return stateMachine.toString();
```

Comment:
Doesn't seem ideal to have only field in the `toString` even though this class has many other fields in it. Maybe provide a getter for `stateMachine` and then log that ?

Reply:
> Doesn't seem ideal to have only field in the toString even though this class has many other fields in it. 

this is consistent with `io.trino.execution.SqlStage#toString`.

> Maybe provide a getter for stateMachine and then log that ?

That would require some transformations even when `debug` logging is off